nodes/modifier_make/cut_object_by_surface.py

```diff
@@ -220,22 +259,25 @@ def remember_connection(edges_by_face, bm_edge, bm_intersection):
                             edges_to_remove.append((old_edge.verts[0], old_edge.verts[1]))
                             need_split = True
                             new_vert_1 = new_verts_by_old_edge[old_edge.index]
-                            self.debug("Old edge %s - %s is to be split; use new vert %s", old_vert_1.index, old_vert_2.index, new_vert_1.index)
+                            # self.debug("Old edge %s - %s is to be split; use new vert %s", old_vert_1.index, old_vert_2.index, new_vert_1.index)
                             new_face_verts[new_face_side].append(new_vert_1)
                             # if this edge is to be split, then we are stepping from one piece of
                             # the face to another
                             new_face_side = not new_face_side
                             new_face_verts[new_face_side].append(new_vert_1)
                             new_face_verts[new_face_side].append(old_vert_2)
                         else:
-                            self.debug("Old edge %s - %s is not to be split", old_vert_1.index, old_vert_2.index)
+                            # self.debug("Old edge %s - %s is not to be split", old_vert_1.index, old_vert_2.index)
                             new_face_verts[new_face_side].append(old_vert_2)
 
                     if need_split:
                         faces_to_remove.append(old_obj_face)
-                        self.debug("Splitting face: %s", [v.index for v in old_obj_face.verts])
+                        # self.debug("Splitting face: %s", [v.index for v in old_obj_face.verts])
                         for new_face_list in new_face_verts.values():
-                            bm_obj.faces.new(new_face_list)
+                            try:
+                                bm_obj.faces.new(new_face_list)
+                            except TypeError:
```

Comment:
TypeError seems like a weird exception to expect there.

Reply:
Well, it is an error that occured during testing of the node, and since all of the code below is meaningless in case of multiple exceptions, I only need the node to ignore everything but detecting intersections. This may not be the best way to do it, but I didn't want to touch anything of the code for generating pieces, since I don't understand it well and don't need to remove limitations there. In fact, this part will probably need a massive overhaul to deal with the limitations.

Reply:
i'm also not knee-deep acquainted with the logic of this node so will avoid interfering any further.